internal/services/controller/controller.go

```diff
@@ -267,6 +257,44 @@ func (c *Controller) Run(ctx context.Context) error {
 	return nil
 }
 
+func (c *Controller) startConditionalInformersWithWatcher(ctx context.Context, conditionalInformers []conditionalInformer) {
```

Comment:
Let's add a test for this, where not all apis are available and only those get registered?